### PR TITLE
Add script to check http links in our translation files

### LINKED ISSUE #12040: Extract all docs links from dashboard for upcoming new URL structure
We need to get all the existing documentation links in our UI (Dashboard primarily though I don't know if Ember has anything that we actively use) to the Write Club as some recent rebranding has meant that the URL structure will change.

@nwmac may have a script to extract this quickly already so all we have to provide @gunamata with is a list. He can then provide us with new URL patterns for us to then update for 2.10.0 or 2.10.1 which could spawn a new ticket depending on the effort required.

### PR DESCRIPTION
**Pull request by nwmac**
### Summary
Fixes #12040 

### Occurred changes and/or fixed issues

- Adds a script which checks the doc http/s links for broken links
- Fixes 3 links that ended in /, which caused a 404 error - fix is to remove trailing slash
- Fixes 3 links by using the correct link in the docs
- Updated the test workflow to run the check script so that this is done as part of the PR workflow

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: scripts/check-i18n-links [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+let i18n = loadI18nFiles(srcFolder);
+
+i18n = { ...i18n, ...loadI18nFiles(pkgFolder) };
+
+console.log(`Read  ${cyan}${ Object.keys(i18n).length }${reset} translations`); // eslint-disable-line no-console
+
+const links = [];
+
+// Look for translations that are not used
```
Comment (richard-cox) [inline scripts/check-i18n-links:174]: Low priority, comment incorrect

File: scripts/check-i18n-links [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+Object.keys(i18n).forEach((str) => {
+  const link = parseLinks(i18n[str]);
+
+ links.push(...link);
+});
+
+console.log(`Discoverd ${cyan}${links.length}${reset} links`); // eslint-disable-line no-console
```
Comment (aalves08) [inline scripts/check-i18n-links:181]: Small typo: 

```suggestion
console.log(`Discovered ${cyan}${links.length}${reset} links`); // eslint-disable-line no-console
```

File: shell/assets/translations/en-us.yaml [hunk 3]
```diff
@@ -1885,7 +1885,7 @@ cluster:
     header: Registry for Rancher System Container Images
     label: Enable cluster scoped container registry for Rancher system container images
     description: "If enabled, Rancher will pull container images from this registry during cluster provisioning. By default, Rancher will also use this registry when installing Rancher's official Helm chart apps. If the cluster scoped registry is disabled, system images are pulled from the System Default Registry in the global settings."
-    docsLinkRke2: "For help configuring private registry mirrors, see the RKE2 <a href=\"https://docs.rke2.io/install/containerd_registry_configuration/\" target=\"_blank\">documentation.</a>"
+    docsLinkRke2: "For help configuring private registry mirrors, see the RKE2 <a href=\"https://docs.rke2.io/install/containerd_registry_configuration\" target=\"_blank\">documentation.</a>"
```
Comment (richard-cox) [inline shell/assets/translations/en-us.yaml:1888]: Low priority, it works, but there's a trailing `\` in href

**Review by richard-cox — APPROVED** [review]
This could block all PRs when docs randomly change, rather than just picking up new link that fail. Just to confirm, the quick fix would be to add `-x` to the test workflow?

There's only two tiny comments, given gates are queued happy to get this in as is

File: scripts/check-i18n-links (hunk 2 above)
Comment (nwmac) [reply scripts/check-i18n-links]: Fixed

File: shell/assets/translations/en-us.yaml (hunk 3 above)
Comment (nwmac) [reply shell/assets/translations/en-us.yaml:1888]: @richard-cox That slash is escaping the double-quote - unless I'm looking at the wrong thing?

File: scripts/check-i18n-links (hunk 1 above)
Comment (nwmac) [reply scripts/check-i18n-links]: Fixes

**Review by aalves08 — APPROVED** [review]
LGTM

**Review by richard-cox — APPROVED** [review]

**Pull request merged**